<?xml version="1.0"?>
<robot name="Human">
  <!-- Links -->
  <link name="base_link" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/Body.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="Neck" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/Neck.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="Head" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/Head.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="RightArm1" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/RightArm1.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="RightArm2" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/RightArm2.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="RightHand1" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/RightHand1.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="LeftArm1" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/LeftArm1.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="LeftArm2" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/LeftArm2.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="LeftHand1" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/LeftHand1.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="RightLeg1" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/RightLeg1.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="RightLeg2" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/RightLeg2.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="RightFoot" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/RightFoot.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="LeftLeg1" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/LeftLeg1.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="LeftLeg2" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/LeftLeg2.dae"/>
      </geometry>
    </visual>
  </link>

  <link name="LeftFoot" >
    <visual>
      <geometry>
        <mesh filename="package://tms_ss_kinect_v2/model/male/LeftFoot.dae"/>
      </geometry>
    </visual>
  </link>

  <!-- Joints -->
  <joint name="CervicalSpine" type="continuous">
    <parent link="Neck"/>
    <child link="Head"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="ShoulderLeft" type="continuous">
    <parent link="base_link"/>
    <child link="LeftArm1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="ElbowLeft" type="continuous">
    <parent link="LeftArm1"/>
    <child link="LeftArm2"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="WristLeft" type="continuous">
    <parent link="LeftArm2"/>
    <child link="LeftHand1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="ShoulderRight" type="continuous">
    <parent link="base_link"/>
    <child link="RightArm1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="ElbowRight" type="continuous">
    <parent link="RightArm1"/>
    <child link="RightArm2"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="WristRight" type="continuous">
    <parent link="RightArm2"/>
    <child link="RightHand1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="HipLeft" type="continuous">
    <parent link="base_link"/>
    <child link="LeftLeg1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="KneeLeft" type="continuous">
    <parent link="LeftLeg1"/>
    <child link="LeftLeg2"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="AnkleLeft" type="continuous">
    <parent link="LeftLeg2"/>
    <child link="LeftFoot"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="HipRight" type="continuous">
    <parent link="base_link"/>
    <child link="RightLeg1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="KneeRight" type="continuous">
    <parent link="RightLeg1"/>
    <child link="RightLeg2"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="AnkleRight" type="continuous">
    <parent link="RightLeg2"/>
    <child link="RightFoot"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="SpineShoulder" type="continuous">
    <parent link="base_link"/>
    <child link="Neck"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

</robot>
